src/components/RightSidebar/TableOfContents.tsx

```diff
@@ -126,13 +126,13 @@ const TableOfContents: FunctionalComponent<Props> = ({ toc = [], labels, isMobil
 	return (
 		<Container>
 			<HeadingContainer>
-				<h2 class="heading" id={onThisPageID}>
+				<h2 className="heading" id={onThisPageID}>
 					{labels.onThisPage}
 				</h2>
 			</HeadingContainer>
-			<ul class="toc-root">
-				{toc.map((heading) => (
-					<TableOfContentsItem key={heading.slug} heading={heading} />
+			<ul className="toc-root">
+				{toc.map((heading2) => (
```

Comment:
This one is a hack as `ComponentChildren` from `preact` seems to mess something with `eslint-plugin-react`. I've set up a reproducible example here https://stackblitz.com/edit/node-zkuutt?file=index.tsx,tsconfig.json,package.json,types.ts,package-lock.json,node_modules%2F%40astrojs%2Fmarkdown-remark%2Fdist%2Ftypes.d.ts